astroid/interpreter/_import/spec.py

```diff
@@ -211,13 +211,20 @@ class ZipFinder(Finder):
 
     def __init__(self, path: Sequence[str]) -> None:
         super().__init__(path)
-        self._zipimporters = _precache_zipimporters(path)
+        for entry_path in path:
```

Comment:
It seems we're still pre-caching, maybe we could add to cache only when required with a subclass and a property ?

```python
class CachedZipImporters:

    def __init__(self):
        self.cache = {}

    def __getattr__(path):
        if path not in self.cache:
            self.cache[path] = zipimport.zipimporter(path)
        return self.cache[path]
```
This way we cache only when we actually need the value ?